lib/zendesk_api/middleware/response/zendesk_request_event.rb

```diff
@@ -0,0 +1,59 @@
+require "faraday/response"
+
+module ZendeskAPI
+  module Middleware
+    module Response
+      # @private
+      class ZendeskRequestEvent < Faraday::Middleware
+        def initialize(app, options = {})
+          super(app)
+          @instrumentation = options[:instrumentation]
+          @logger = options[:logger]
+        end
+
+        def call(env)
+          start_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+          @app.call(env).on_complete do |response_env|
+            end_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+            duration = (end_time - start_time) * 1000.0
+            if @instrumentation
```

Comment:
Nit: this method has a lot of nested ifs. Should some of these layers become something like `next unless @instrumentation`?